http/httpclient/src/main/java/com/okta/commons/http/httpclient/HttpClientRequestExecutor.java

```diff
@@ -263,17 +255,98 @@ private HttpHeaders getHeaders(HttpResponse response) {
+        Integer configuredValue = this.requestExecutorParams.get(key);
+        if (configuredValue != null) {
+            return configuredValue;
+        }
+        log.warn("Failed to read configuration property [{}: {}]. Falling back to default value: {}", key, warning, defaultValue);
+        return defaultValue;
+    }
 
-        int configuredValue = defaultValue;
-        String configuredValueString = System.getProperty(key);
-        if (configuredValueString != null) {
-            try {
-                configuredValue = Integer.parseInt(configuredValueString);
-            } catch (NumberFormatException nfe) {
-                log.warn("{}: {}. Using default: {}.", warning, configuredValueString, defaultValue, nfe);
-            }
+    int getMaxConnectionPerRoute() {
+        return getRequestExecutorParam(
+            "maxConnectionsPerRoute",
+            "Bad max connection per route value",
+            DEFAULT_MAX_CONNECTIONS_PER_ROUTE
+        );
+    }
+
+    int getMaxConnectionTotal() {
+        return getRequestExecutorParam(
+            "maxConnectionsTotal",
+            "Bad max connection total value",
+            DEFAULT_MAX_CONNECTIONS_TOTAL
+        );
+    }
+
+    int getMaxConnectionInactivity() {
+        return getRequestExecutorParam(
+            "validateAfterInactivity",
+            "Invalid max connection inactivity validation value",
+            DEFAULT_CONNECTION_VALIDATION_INACTIVITY
+        );
+    }
+
+    int getConnectionTimeToLive() {
+        return getRequestExecutorParam(
+            "timeToLive",
+            "Invalid connection time to live value",
+            DEFAULT_CONNECTION_TIME_TO_LIVE
+        );
+    }
+
+    private void parseRequestExecutorParams(Map<String, String> props) {
+
+        Integer maxPerRoute = lookupConfigValue(
+            props,
+            "maxConnectionsPerRoute",
+            MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY);
+        if(maxPerRoute != null) {
+            this.requestExecutorParams.put("maxConnectionsPerRoute", maxPerRoute);
+        }
+
+        Integer maxTotal = lookupConfigValue(
+            props,
+            "maxConnectionsTotal",
+            MAX_CONNECTIONS_TOTAL_PROPERTY_KEY);
+        if(maxTotal != null) {
+            requestExecutorParams.put("maxConnectionsTotal", maxTotal);
+        }
+
+        Integer validateAfterInactivity = lookupConfigValue(
+            props,
+            "validateAfterInactivity",
+            CONNECTION_VALIDATION_PROPERTY_KEY);
+        if(validateAfterInactivity != null) {
+            requestExecutorParams.put("validateAfterInactivity", validateAfterInactivity);
+        }
+
+        Integer timeToLive = lookupConfigValue(
+            props,
+            "timeToLive",
+            CONNECTION_TIME_TO_LIVE_PROPERTY_KEY);
+        if(timeToLive != null) {
+            requestExecutorParams.put("timeToLive", timeToLive);
```

Comment:
nit: this prop name should be more descriptive there are multiple types of TTLs, maybe `connectionTimeToLive` ?

Reply:
updated